ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using QuantConnect.Logging;
+
+namespace QuantConnect.ToolBox.PsychSignalDataConverter
+{
+    public class PsychSignalDataConverter
+    {
+        /// <summary>
+        /// Iterate over the data, processing each data point before finally zipping the directories
+        /// </summary>
+        public void Convert(string sourceFilePath, SecurityType securityType, string market)
+        {
+            if (securityType != SecurityType.Equity)
+            {
+                throw new ArgumentException("Only equity data is supported for the psychsignal converter. Exiting...");
+            }
+
+            var tickerFileHandlers = new Dictionary<string, TickerData>();
+            var tickerFolders = new List<string>();
+
+            var dataFolder = Path.Combine(Globals.DataFolder, securityType.SecurityTypeToLower(), market);
+            var sentimentFolder = Path.Combine(dataFolder, "alternative", "psychsignal");
```

Comment:
We construct this path in several places, consider adding a static method to the base data type to construct a full path and update other usages to call the static method, something like `PsychSignalSentimentData.CreatePath(string dataFolder)` ?